pkg/startup/startup.go

```diff
@@ -555,7 +555,11 @@ func configureIPsAndSubnets(node *api.Node) (bool, error) {
 
 	if ipv4Env == "none" && (ipv6Env == "" || ipv6Env == "none") && node.Spec.BGP.IPv4Address == "" && node.Spec.BGP.IPv6Address == "" {
 		log.Warn("No IP Addresses configured, and autodetection is not enabled")
-		terminate()
+		// Unrecoverable if Calico is handling networking, terminate to restart.
```

Comment:
Not fully sure I follow the check on line 556.

Maybe we could move this block out into the calling function as part of the post-configureIPs check.  In the calling function I think you'd just need to check that NETWORKING BACKEND is none and both IPs are blank.  If so terminate.

I don't think it's necessary to check the ipv4Env and ipv6Env values are none or blank.  If they are not set to none (or blank) then we'll hit an error if we don't manage to extract an IP from the value, and we only really care if we don't have any IPs and we are using calico networking.

Reply:
I would read :556 as neither IPv4 or IPv6 addresses are available under calico/node resource.

Do you mean? do the following after calling `configureIPsAndSubnets`:

```
if node.Spec.BGP.IPv4Address == "" && node.Spec.BGP.IPv6Address == "" {
    if os.Getenv("CALICO_NETWORKING_BACKEND") != "none" {
        // log-1
	terminate()
     } else {
        // log-2
     } 
}
```

if yes, question in my earlier comment is relevant to this proposal.

Thanks

Reply:
Yup that's what I was thinking.  It feels easier to following having the "assign IPs" separate from the "check IP config is consistent with other config".